cjworkbench/tests/i18n/test_trans.py

```diff
@@ -22,8 +23,26 @@ def test_trans_params(self):
             "Hello you there {c}!",
         )
 
+    # Tests that a programmer will break the system by including an invalid parameter
+    def test_programmer_invalid_parameter_syntax(self):
+        with self.assertRaises(ICUError):
+            trans(
+                mock_message_id,
+                default="Hello {a b}",
+                parameters={"a": "you", "b": "2"},
+            )
+
+    # Tests that a translator can't break our system by including an invalid parameter
+    def test_translator_invalid_parameter_syntax(self):
+        self.assertEqual(
+            MessageTranslator(default_locale)._process_simple_message(
+                "Hello {a b}", "Hello {a} {b}", parameters={"a": "you", "b": "!"}
+            ),
+            "Hello you !",
+        )
+
     # Tests that a programmer can include a numeric variable in the message
-    def test_format_invalid_default(self):
+    def test_programmer_numeric_parameter(self):
```

Comment:
Let's not test any distinction between "programmer" and "translator". `test_programmer_numeric_parameter()` and `test_translator_numeric_parameter()` do the exact same thing (though one calls a private method and the other doesn't).

_Every_ test involves both a programmer and a translator:

* `parameters` is set by a programmer.
* `message` is set by a translator
* `default` is set by ... who cares? (I hesitate to call the its author either "programmer" _or_ "translator".)

Don't test by "roles": test by _arguments_:

* Invalid `parameters` should raise an uncaught exception
* Invalid `message` should fallback to default (and we'll probably want to log the failure in the near future)
* Invalid `default` should raise an uncaught exception